src/main/java/org/assertj/core/api/AbstractIterableAssert.java

```diff
@@ -677,6 +678,64 @@ public SELF usingDefaultElementComparator() {
   }
 
   /**
+   * Verifies that the actual {@link Iterable} contains at least one of the given elements, in any order.
+   * <p>
+   * Example :
+   * <pre><code class='java'> Iterable&lt;String&gt; abc = Arrays.asList("a", "b", "c");
+   *
+   * // assertions will pass
+   * assertThat(abc).containsAnyOf("b")
+   *                .containsAnyOf("b", "c")
+   *                .containsAnyOf("a", "b", "c")
+   *                .containsAnyOf("a", "b", "c", "d")
+   *                .containsAnyOf("e", "f", "g", "b");
+   *
+   * // assertions will fail
+   * assertThat(abc).containsAnyOf("d");
+   * assertThat(abc).containsAnyOf("d", "e", "f", "g");</code></pre>
+   *
+   * @param values the values that, at least one of which is expected to be in the given {@code Iterable}.
+   * @return {@code this} assertion object.
+   * @throws NullPointerException if the array of values is {@code null}.
+   * @throws IllegalArgumentException if the array of values is empty and given {@code Iterable} is not empty.
+   * @throws AssertionError if the given {@code Iterable} is {@code null}.
+   * @throws AssertionError if the given {@code Iterable} does not contain any of given {@code values}.
+   */
+  public SELF containsAnyOf(ELEMENT... values) {
+    iterables.assertContainsAnyOf(info, actual, values);
+    return myself;
+  }
+
+  /**
+   * Verifies that the actual {@link Iterable} contains at least one of the given elements, in any order.
+   * <p>
+   * Example :
+   * <pre><code class='java'> Iterable&lt;String&gt; abc = Arrays.asList("a", "b", "c");
+   *
+   * // assertions will pass
+   * assertThat(abc).containsAnyElementsOf(Arrays.asList("b"))
+   *                .containsAnyElementsOf(Arrays.asList("b", "c"))
+   *                .containsAnyElementsOf(Arrays.asList("a", "b", "c"))
+   *                .containsAnyElementsOf(Arrays.asList("a", "b", "c", "d"))
+   *                .containsAnyElementsOf(Arrays.asList("e", "f", "g", "b"));
+   *
+   * // assertions will fail
+   * assertThat(abc).containsAnyElementsOf(Arrays.asList("d"));
+   * assertThat(abc).containsAnyElementsOf(Arrays.asList("d", "e", "f", "g"));</code></pre>
+   *
+   * @param iterable the iterable of values that, at least one of which is expected to be in the given {@code Iterable}.
+   * @return {@code this} assertion object.
+   * @throws NullPointerException if the iterable of expected values is {@code null}.
+   * @throws IllegalArgumentException if the iterable of expected values is empty.
+   * @throws AssertionError if the given {@code Iterable} is {@code null}.
+   * @throws AssertionError if the given {@code Iterable} does not contain any of elements from other given {@code Iterable}.
+   */
+  public SELF containsAnyElementsOf(Iterable<ELEMENT> iterable) {
+    iterables.assertContainsAnyElementsOf(info, actual, iterable);
+    return myself;
```

Comment:
it is simpler to call `containsAnyOf` like this: `return containsAnyOf(toArray(iterable));` then the only test you need for this guys is that it delegates correctly to `iterables.assertContainsAnyOf`.
You can also get rid of `iterables.assertContainsAnyElementsOf` and its corresponding tests.